### PR TITLE
The culling: nix boto SupportReasonAPIStorage

### PR DESCRIPTION
**Pull request by willkg**
We're not using this anymore and large portions of it are commented out.

r?

### CODE REVIEW

**Comment by willkg** [comment]
So far, the culling is comprised of 400 lines added and 3300 lines removed. There's a lot of low-risk obvious changes in there, but then a bunch of more complicated changes where I'm squashing down class hierarchies and making certain things consistent across the codebase. I figure it's easier on the reviewer if I break it up into multiple PRs and that'll make it more likely issues get caught.

This is the first in a series of "culling things from the collector" PRs.

@adngdb Can you look through this? I think reviewing probably involves:

1. Do the changes make sense?
2. Did I miss anything?
3. Did Travis smile upon us?

File: collector/unittest/external/boto/test_crashstorage.py [hunk 1]
```diff
@@ -147,8 +146,8 @@ def setup_mocked_s3_storage(
         if storage_class == 'BotoS3CrashStorage':
             config.bucket_name = 'crash_storage'
             s3 = BotoS3CrashStorage(config)
-        elif storage_class == 'SupportReasonAPIStorage':
-            s3 = SupportReasonAPIStorage(config)
+        else:
+            raise Exception('No such storage_class')
```
Comment (adngdb) [inline collector/unittest/external/boto/test_crashstorage.py]: It doesn't sound good to raise ``Exception``. How about ``NotImplementedError`` instead?

**Comment by adngdb** [comment]
r+ with a nit.

File: collector/unittest/external/boto/test_crashstorage.py (hunk 1 above)
Comment (willkg) [inline collector/unittest/external/boto/test_crashstorage.py]: This is in the test infrastructure so I don't think the exception class raised makes a difference.

The better fix is probably to remove the code that allows us to switch on the `storage_class` at all. I'll look at doing that.

**Comment by willkg** [comment]
^^^ That nixed that bit of code altogether thus alleviating the nit. Travis is happy, so I'm going to land this.

Thank you!

**Pull request merged**